src/include/optimizer/operator_node.h

```diff
@@ -67,39 +105,55 @@ class OperatorNode {
    * Move constructor
    * @param op other to construct from
    */
-  OperatorNode(OperatorNode &&op) noexcept : op_(std::move(op.op_)), children_(std::move(op.children_)) {}
+  OperatorNode(OperatorNode &&op) noexcept : contents_(op.contents_), children_(std::move(op.children_)) {}
 
   /**
    * @return vector of children
    */
-  std::vector<common::ManagedPointer<OperatorNode>> GetChildren() const {
-    std::vector<common::ManagedPointer<OperatorNode>> result;
+  std::vector<common::ManagedPointer<AbstractOptimizerNode>> GetChildren() const override {
+    std::vector<common::ManagedPointer<AbstractOptimizerNode>> result;
     result.reserve(children_.size());
-    for (auto &i : children_) result.emplace_back(i);
+    int count = 0;
```

Comment:
What is `count` tracking?

Reply:
ah shoot, I think I had that for debugging purposes when working out the memory leak, but it doesn't do anything for the functionality. I'll get rid of it